services/stripe-webhook.php

```diff
@@ -412,19 +413,62 @@
 				pmpro_stripeWebhookExit();
 			}
 		}
-	}
-	elseif( $pmpro_stripe_event->type == "charge.refunded" )
-	{
+		elseif( $pmpro_stripe_event->type == "charge.refunded" )
+		{
+		
+		/**
+		 * Stripe's charge.refunded takes a while to come through. 
+		 *
+		 * When requesting a refund, you'll have a payment_intent.succeeded first,
+		 * and then the charge.refunded will come through later.
+		 */
+
+		/**
+		 * ch_ prefixed. Using this in case someone tries to resend a webhook, then we get this ID
+		 * instead of an event ID
+		 */
+		$payment_transaction_id = $pmpro_stripe_event->data->object->id;
+
+		$morder = new MemberOrder();
 
-		//Lets handle refunds here
+		$morder->getMemberOrderByPaymentTransactionID( $payment_transaction_id );		
 
-		$payment_transaction_id = $pmpro_stripe_event->id;
+		//We've got the right order	
+		if( !empty( $morder->id ) ) { 
+				
+			$morder->status = 'refunded';
 
-		$morder = new MemberOrder();
+			// translators: %1$s is the date of the refund. %2$s is the transaction ID.
+			
+			$logstr .= sprintf( __('Order successfully refunded on %1$s for transaction ID %2$s at the gateway.', 'paid-memberships-pro' ), date_i18n('Y-m-d H:i:s'), $payment_transaction_id );	
+			//Add to order notes. 
+			
+			// translators: %1$s is the date of the refund. %2$s is the transaction ID.
+			$morder->notes = trim( $morder->notes ) .' '. sprintf( __('Order successfully refunded on %1$s for transaction ID %2$s at the gateway.', 'paid-memberships-pro' ), date_i18n('Y-m-d H:i:s'), $payment_transaction_id );
```

Comment:
The `trim()` should wrap the whole new note and not just the previous note.

```suggestion
			$morder->notes = trim( $morder->notes . ' ' . sprintf( __('Order successfully refunded on %1$s for transaction ID %2$s at the gateway.', 'paid-memberships-pro' ), date_i18n('Y-m-d H:i:s'), $payment_transaction_id ) );
```